gno.land/pkg/sdk/vm/gas.go

```diff
@@ -0,0 +1,48 @@
+package vm
+
+import (
+	"fmt"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/overflow"
+)
+
+const (
+
+	// We will use gasMemFactor and gasCpuFactor to multiply the vm memory allocation and cpu cycles to get the gas number.
+	// We can use these two factoctors to keep the gas for storage access, CPU and Mem in reasonable proportion.
+	gasFactorMem int64 = 1 // change this value based on gas profiling
+	gasFactorCpu int64 = 1 // change this value based on gas profiling
+
+	logPrefixAddPkg   = "gas.vm.addpkg"
+	logPrefixCall     = "gas.vm.call"
+	logPrefixRun      = "gas.vm.run"
+	logPrefixQeval    = "gas.vm.qeval"
+	logPrefixQevalStr = "gas.vm.qevalstr"
+)
+
+// consume gas and log vm gas usage
+func consumeGas(ctx sdk.Context, m *gno.Machine, prefix string, pkgPath string, expr string) {
+	_, mem := m.Alloc.Status()
+
+	gasCpu := overflow.Mul64p(m.Cycles, gasFactorCpu)
+	gasMem := overflow.Mul64p(mem, gasFactorMem)
+
+	// we simplify the log here, the storage gas log included tx size and sigature verification gas.
+	storeLog := fmt.Sprintf("%s.storage, %s %s, %d", prefix, pkgPath, expr, ctx.GasMeter().GasConsumed())
```

Comment:
```suggestion
	storeLog := fmt.Sprintf("%s.txSize/signatures/storage, %s %s, %d", prefix, pkgPath, expr, ctx.GasMeter().GasConsumed())
```

To avoid confusion, I think it's better to mention all gas consumption, even if later this is improved.